### DISCUSSION TITLE
Forking Renovate removes labels from Dependency Dashboard

### DISCUSSION THREAD

**Discussion by Neenu1995** [renovatebot/renovate#10798] [Request Help]
Since 9th July, Forking Renovate keeps removing labels from Dependency Dashboard issues on repos in googleapis org. Looks like most of the issues were created and labelled on or around September 23, 2020. Even after relabelling them once, a few issues had labels removed again.

Example issues with labels removed after relabelling:
- [java-iamcredentials](https://github.com/googleapis/java-iamcredentials/issues/226 )
- [java-billing](https://github.com/googleapis/java-billing/issues/186)

Example issues with labels removed once:
- [java-translate](https://github.com/googleapis/java-translate/issues/287)
- [java-language](https://github.com/googleapis/java-language/issues/232)

@danoscarmike

**Comment by Neenu1995** [comment]
Looks like labels are being removed from libraries after a release. Released 3 libraries today. All of them had labels removed from Dependency dashboard issue.
This could be true for the earlier libraries too, because there was a bulk release just before the problem emerged. Not 100% sure that all libraries that are being released encounter this issue.

**Reply by viceice** [reply]
I think #10601 feature is the cause. Can you try `dependencyDashboardLabels` to explicit set your dashboard labels as workaround?


https://docs.renovatebot.com/configuration-options/#dependencydashboardlabels

**Reply by viceice** [reply]
I think we need to check if labels are set and only send if not `undefined`

https://github.com/renovatebot/renovate/blob/ac4f4a3f6da9c0d35883e9a0a02f60d3843641b7/lib/platform/github/index.ts#L1205-L1212

It has maybe broken other platforms too. Will try to fix asap.

**Reply by rarkins** [reply]
Yeah, it should ignore the labels unless the array is non empty

**Reply by viceice** [reply]
I think we should use `undefined` and `null` as ignore and all other values as override. so an empty array should remove all labels. 🤔

**Reply by viceice** [reply]
pr to fix at #10806

**Reply by Neenu1995** [reply]
Thank you so much for the speedy resolution :)

**Reply by rarkins** [reply]
FYI I haven't deployed it to the app so far but should be within 1-2 days

**Reply by Neenu1995** [reply]
Sounds Good! Thank you.

**Reply by Neenu1995** [reply]
Hi folks,
Is the app deployed yet?

**Reply by Neenu1995** [reply]
Looks like it is still removing labels.
https://github.com/googleapis/java-orchestration-airflow/issues/9

Is there any action required from our side?

**Reply by rarkins** [reply]
Can you check for log messages for the job/run when the labels were removed? Confirm the renovate version and if anything was logged towards the end of the run when the dashboard is allied.

**Reply by Neenu1995** [reply]
We don't run it ourselves and don't have any logs for it.

**Reply by Neenu1995** [reply]
Let me see if I can run the cli manually and get that info.

**Reply by rarkins** [reply]
Don't worry, I'll try checking the app logs.

**Reply by viceice** [reply]
BTW: the hosted app logs are avauialable at https://app.renovatebot.com/dashboard for repo maintainer 🙃

**Reply by rarkins** [reply]
I think it unfortunately doesn't work for forking renovate

**Reply by rarkins** [reply]
Version is ok:

```
"renovateVersion":"25.56.5"
```

Logs don't give anything away:

```
{"level":20,"msg":"Ensuring Dependency Dashboard","time":"2021-07-21T17:10:46.425Z"}
{"level":30,"result":"no-work","msg":"Blocked PR","time":"2021-07-21T17:10:46.425Z"}
{"level":30,"result":"no-work","msg":"Blocked PR","time":"2021-07-21T17:10:46.426Z"}
{"level":20,"msg":"ensureIssue(Dependency Dashboard)","time":"2021-07-21T17:10:46.629Z"}
{"level":20,"msg":"Patching issue","time":"2021-07-21T17:10:46.630Z"}
{"level":20,"msg":"Issue updated","time":"2021-07-21T17:10:47.405Z"}
```

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/pull/10806/files#diff-972d6bae74954809c949c571a8cc9fddb7f634d8a164b71059d679f4e2f822d7R1206

**Comment by JustinBeckwith** [comment]
Greetings folks!  This is causing us a little pain with various tools/dashboards that use labels for team tracking.  Anything we can do here to help?

**Reply by viceice** [reply]
I'll try to fix it tomorrow

**Reply by rarkins** [reply]
I think it's a one-liner so I'll fix now

**Reply by rarkins** [reply]
I'll put https://github.com/renovatebot/renovate/pull/10976 live tomorrow

**Reply by rarkins** [reply]
Now done

**Reply by JustinBeckwith** [reply]
Thank you so much!
